topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -406,12 +413,62 @@ private TopicStore createTopicStore(Zk zk, Config config) {
     }
 
     private void setSaslConfigs(Properties kafkaClientProps) {
-        String saslMechanism;
-        String jaasConfig;
+        String configSaslMechanism = config.get(Config.SASL_MECHANISM);
+        
+        switch (configSaslMechanism) {
+            case SASL_TYPE_PLAIN:
+            case SASL_TYPE_SCRAM_SHA_256:
+            case SASL_TYPE_SCRAM_SHA_512:
+                setStandardSaslConfigs(kafkaClientProps);
+                break;
+            case "custom":
+                setCustomSaslConfigs(kafkaClientProps);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid SASL_MECHANISM: '" + configSaslMechanism + "'");
+        }
+    }
+
+    private void setCustomSaslConfigs(Properties kafkaClientProps) {
+        String customPropsString = config.get(Config.SASL_CUSTOM_CONFIG);
+
+        if (customPropsString.isEmpty()) {
+            throw new InvalidConfigurationException("Custom SASL config properties are not set");
```

Comment:
You should include the value of `Config.SASL_CUSTOM_CONFIG` in the message so the user know exactly what it is that they haven't set. It also makes it easier for them to find the relevant bit in the docs.

```suggestion
            throw new InvalidConfigurationException(Config.SASL_MECHANISM + "=custom, but env var " + Config.SASL_CUSTOM_CONFIG + " is not set");
```